packages/payment/src/models/payment-session.ts

```diff
@@ -65,10 +72,52 @@ export default class PaymentSession {
   @BeforeCreate()
   onCreate() {
     this.id = generateEntityId(this.id, "payses")
+
+    const val = new BigNumber(this.raw_amount ?? this.amount)
+
+    this.amount = val.numeric
+    this.raw_amount = val.raw!
+  }
+
+  get amount_() {
+    return new BigNumber(this.raw_amount ?? this.amount).numeric
+  }
+
+  @BeforeUpdate()
+  onUpdate() {
+    const val = new BigNumber(this.raw_amount ?? this.amount)
+
+    console.log("!!! ON BEFORE UPDATE", typeof this.amount)
+    this.amount = val.numeric
+    this.raw_amount = val.raw!
+  }
+
+  @AfterUpdate()
+  onAfterUpdate() {
+    const val = new BigNumber(this.raw_amount ?? this.amount)
+
+    console.log("!!! ON AFTER UPDATE", typeof this.amount)
+    this.amount = val.numeric
+    this.raw_amount = val.raw!
   }
 
   @OnInit()
   onInit() {
     this.id = generateEntityId(this.id, "payses")
+
+    console.log("!!! ON INIT", typeof this.amount)
+
+    const amount = this.raw_amount ?? this.amount
+
+    // OnInit can be sometimes fired twice, once when an entity reference is created, and once after its populated.
+    // The first call will only contain the id and amounts will be undefined so calling the BigNumber constructor will fail.
+    // On the second call with populated data amounts will exist.
+    if (typeof amount === "undefined") {
+      return
+    }
```

Comment:
@riqwan, this does sound an awful lot like what you experienced no?

Reply:
We should probably move BigNumber stuff from `@OnInit` to `@OnLoad` which will only fire when the entity is fully loaded.